### ISSUE TITLE
@nx/dependency-checks --fix ignores dev dependencies in @nx/js library with vitest

### ISSUE THREAD

**Issue by arthurgubaidullin** [nrwl/nx#18874]
### Current Behavior

`@nx/dependency-check` --fix always adds the following dependencies to package.json:
```json
{
  "dependencies": {
    "vite": "~4.3.9",
    "@nx/vite": "16.7.4"
  },
  "devDependencies": {
    "vite": "~4.3.9",
    "@nx/vite": "16.7.4"
  },
}
```

This pollutes the built build.

### Expected Behavior

If dev dependencies are added, they should not be re-added.

### Steps to Reproduce

1. Add @nx/js library with vitest
2. Run `npx nx run-many --target build,test,lint --fix`
3. Check out `package.json`

### Nx Report

```shell
>  NX  Falling back to ts-node for local typescript execution. This may be a little slower.
  - To fix this, ensure @swc-node/register and @swc/core have been installed

>  NX   Report complete - copy this into the issue template

   Node   : 18.16.1
   OS     : linux-arm64
   npm    : 9.5.1
   
   nx                 : 16.7.4
   @nx/js             : 16.7.4
   @nx/linter         : 16.7.4
   @nx/workspace      : 16.7.4
   @nx/cypress        : 16.7.4
   @nx/devkit         : 16.7.4
   @nx/eslint-plugin  : 16.7.4
   @nx/react          : 16.7.4
   @nrwl/tao          : 16.7.4
   @nx/vite           : 16.7.4
   @nx/web            : 16.7.4
   @nx/webpack        : 16.7.4
   typescript         : 5.1.6
```

### Package Manager Version

npm 9.5.1

### Operating System

- [ ] macOS
- [X] Linux
- [ ] Windows
- [ ] Other (Please specify)

### Additional Information

In Nx version 16.6.0 this behavior is not observed.

**Comment by meeroslav** [comment]
The `devDependencies` are ignored since this is not something that should exist in a publishable library. The second problem is that `vite.config.ts` is parsed for dependencies, while it should have been ignored.

This should be fixed with https://github.com/nrwl/nx/pull/18882; unfortunately, it seems that migration is not picking it up correctly.

cc @jaysoo

**Comment by meeroslav** [comment]
It works, but it will be published in the upcoming `16.8.0` version.

**Comment by arthurgubaidullin** [comment]
@meeroslav Thanks. What I meant was that `lint --fix` adds dependencies to the package.json of the source code (not the build), even if they are installed as development dependencies.

**Comment by arthurgubaidullin** [comment]
I confirm. Fixed in Nx 16.8.0.

**Comment by jensbodal** [comment]
What does the config look like that fixes this?
